packages/app/src/cli/services/extensions/bundle.test.ts

```diff
@@ -0,0 +1,213 @@
+    }
+    const stderr: any = {
+      write: vi.fn(),
+    }
+    const app = testApp({
+      directory: '/project',
+      dotenv: {
+        path: '/project/.env',
+        variables: {
+          FOO: 'BAR',
+        },
+      },
+      extensions: {
+        ui: [extension],
+        theme: [],
+        function: [],
+      },
+    })
+    vi.mocked(esBuild).mockResolvedValue(esbuildResultFixture())
+
+    // When
+    await bundleExtension({
+      env: app.dotenv?.variables ?? {},
+      outputBundlePath: extension.outputBundlePath,
+      minify: true,
+      environment: 'production',
+      sourceFilePath: extension.entrySourceFilePath,
+      stdout,
+      stderr,
+    })
+
+    // Then
+    const call = vi.mocked(esBuild).mock.calls[0] as any
+    expect(call).not.toBeUndefined()
+    const options: BuildOptions = call[0]
+
+    expect(options.bundle).toBeTruthy()
+    expect(options.entryPoints).toEqual([extension.entrySourceFilePath])
+    expect(options.outfile).toEqual(extension.outputBundlePath)
+    expect(options.sourceRoot).toEqual(path.dirname(extension.entrySourceFilePath))
+    expect(options.loader).toEqual({
+      '.esnext': 'ts',
+      '.js': 'jsx',
+    })
+    expect(options.legalComments).toEqual('none')
+    expect(options.minify).toBeTruthy()
+    expect(options.target).toEqual('es6')
+    expect(options.resolveExtensions).toEqual(['.tsx', '.ts', '.js', '.json', '.esnext', '.mjs', '.ejs'])
+    expect(options.define).toEqual({
+      'process.env.FOO': JSON.stringify('BAR'),
+      'process.env.NODE_ENV': JSON.stringify('production'),
+    })
+    expect(vi.mocked(stdout.write).calls[0][0]).toMatchInlineSnapshot(`
+      "▲ [WARNING] [plugin plugin] warning text
+
+      "
+    `)
+    expect(vi.mocked(stdout.write).calls[0][0]).toMatchInlineSnapshot(`
+      "▲ [WARNING] [plugin plugin] warning text
+
+      "
+    `)
+  })
+
+  test('stops the ESBuild when the abort signal receives an event', async () => {
+    // Given
+    const extension = testUIExtension()
+    const app = testApp({
+      directory: '/project',
+      dotenv: {
+        path: '/project/.env',
+        variables: {
+          FOO: 'BAR',
+        },
+      },
+      extensions: {
+        ui: [extension],
+        theme: [],
+        function: [],
+      },
+    })
+    const stdout: any = {
+      write: vi.fn(),
+    }
+    const stderr: any = {
+      write: vi.fn(),
+    }
+    const esbuildStop: any = vi.fn()
+
+    vi.mocked(esBuild).mockResolvedValue({
+      errors: [],
+      warnings: [],
+      stop: esbuildStop,
+    })
+    const abortController = new abort.Controller()
+
+    // When
+    await bundleExtension({
+      env: app.dotenv?.variables ?? {},
+      outputBundlePath: extension.outputBundlePath,
+      minify: true,
+      environment: 'production',
+      sourceFilePath: extension.entrySourceFilePath,
+      stdout,
+      stderr,
+      watchSignal: abortController.signal,
+    })
+    abortController.abort()
```

Comment:
Don't you need to provide a `watch` function for esBuild to actually keep watching so you can abort it?
In this case `esbuildStop` is being called, but is not stopping anything, right?

Reply:
> Don't you need to provide a watch function for esBuild to actually keep watching so you can abort it?

That's correct. Otherwise the call to `esbuild` blocks until the bundling completes. In this case I'm testing that the `abort` call propagates to `esbuild`, which is mocked.